### ISSUE TITLE
@vue/preload-webpack-plugin seems not working when generating static site

### ISSUE THREAD

**Issue by saike** [freddy38510/quasar-app-extension-ssg#175]
Hello, i use `@vue/preload-webpack-plugin` inside `quasar.conf.js`

```
    const PreloadWebpackPlugin = require('@vue/preload-webpack-plugin');

    chainWebpack (chain) {

        chain.plugin('preload')
          .use(PreloadWebpackPlugin, [{
            rel: 'preload',
            include: 'initial',
            fileBlacklist: [
              /\.map$/,
              /\.js$/
            ]
        }])
      },
```

Dev server preloads fonts and styles perfectly, 
<img width="693" alt="Screen Shot 2565-02-13 at 14 26 57" src="https://user-images.githubusercontent.com/1619473/153743414-6853a5bb-1cf3-413f-99d4-7c1f50504638.png">

but when i build SSG, head links not preloaded.
<img width="853" alt="Screen Shot 2565-02-13 at 14 27 27" src="https://user-images.githubusercontent.com/1619473/153743421-fba19faf-f6ab-4d2c-bd37-8e7b71be0fbf.png">

 Because of this, fonts on the page loads very long time and flashing. Looks ugly.

Is it possible to make preload plugin works with ssg?

**Comment by freddy38510** [comment]
Hello,

`@vue/preload-webpack-plugin` is an extension plugin for [html-webpack-plugin](https://github.com/jantimon/html-webpack-plugin) which is not involved when doing SSR. It is only for the SPA fallback file `404.html`.

There is many ways to achieve what you want in particular with [Quasar boot files](https://quasar.dev/quasar-cli/boot-files) or by using the [onRouteRendered](https://github.com/freddy38510/quasar-app-extension-ssg#onrouterenderedhtml-route-distdir) hook.

But you gave me the idea to add this as a new feature to this app extension. I think i will try to provide the same options than `@vue/preload-webpack-plugin`.

**Comment by freddy38510** [comment]
@saike,

I'm working on it trying to re-implement the [`shouldPreload`](https://github.com/vuejs/vue-ssr-docs/tree/master/docs/api#shouldpreload) and [`shouldPrefetch`](https://github.com/vuejs/vue-ssr-docs/tree/master/docs/api#shouldprefetch) methods to the [SSR renderer](https://github.com/quasarframework/quasar/blob/dev/app-deps/ssr-helpers/create-renderer.js).

This is almost done and should help you to preload font files.

**Comment by saike** [comment]
@freddy38510 great! thank you for your job. Waiting for updates

**Comment by freddy38510** [comment]
New options `shouldPreload` and `shouldPrefetch` [documented here](https://github.com/freddy38510/quasar-app-extension-ssg#shouldpreload) are available in the latest [v3.3.0](https://github.com/freddy38510/quasar-app-extension-ssg/tree/v3.3.0) release.

If you are concerned about performance, I can suggest you to [use svg](https://quasar.dev/vue-components/icon#svg-icons) instead of web fonts for icons.
For fonts related to text, preloading too much of them can have a bad impact on performance.

For better optimization of fonts loading by playing with the [`font-display`](https://developer.mozilla.org/en-US/docs/Web/CSS/@font-face/font-display) css property it can be better to use[ your own custom fonts.](https://quasar.dev/style/typography#add-custom-fonts)

**Comment by saike** [comment]
@freddy38510 thank you for explanation. I will try new version soon
